{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Set environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "source config_star.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# File name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "B_16_S26_L001_R1_001.fastq.gz\n",
      "B_16_S26_L002_R1_001.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "for FASTQ in $(ls $RAW_FASTQS | grep ^B | head -2); do\n",
    "    echo $FASTQ\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "B_16_S26_L001_R1_001.fastq.gz\n",
      "B_16_S26_L001_R1\n",
      "\n",
      "B_16_S26_L002_R1_001.fastq.gz\n",
      "B_16_S26_L002_R1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for FASTQ in $(ls $RAW_FASTQS | grep ^B | head -2); do\n",
    "    FASTQ_BASE=$(basename ${FASTQ} '_001.fastq.gz')\n",
    "    \n",
    "    echo $FASTQ\n",
    "    echo $FASTQ_BASE\n",
    "    echo \"\"\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "for FASTQ in $(ls $RAW_FASTQS | grep ^B | head -2); do\n",
    "    FASTQ_BASE=$(basename ${FASTQ} '_001.fastq.gz')\n",
    "    \n",
    "    echo $FASTQ\n",
    "    echo $FASTQ_BASE\n",
    "    echo \"\"\n",
    "    ##################################################\n",
    "    fastqc --extract $RAW_FASTQS/B_16_S26_L001_R1_001.fastq.gz -o $QC\n",
    "    \n",
    "    fastq-mcf $ADAPTERS \\\n",
    "        $RAW_FASTQS/B_16_S26_L001_R1_001.fastq.gz \\\n",
    "        -q 20 -x 0.5 \\\n",
    "        -o $TRIMMED/B_16_S26_L001_R1_001.trim.fastq.gz\n",
    "        \n",
    "    fastqc --extract $RAW_FASTQS/B_16_S26_L001_R1_001.fastq.gz -o $QC\n",
    "    \n",
    "    STAR \\\n",
    "        --runMode           alignReads  \\\n",
    "        --twopassMode       None        \\\n",
    "        --genomeDir         $GENOME_DIR \\\n",
    "        --readFilesIn       $TRIMMED/B_16_S26_L001_R1_001.trim.fastq.gz \\\n",
    "        --readFilesCommand  gunzip -c                                   \\\n",
    "        --outFileNamePrefix ${STAR_OUT}/B_16_S26_L001_R1_short_introns_ \\\n",
    "        --quantMode         GeneCounts              \\\n",
    "        --outSAMtype        BAM SortedByCoordinate  \\\n",
    "        --alignIntronMax    5000                    \\\n",
    "        --outSJfilterIntronMaxVsReadN 500 1000 2000\n",
    "        \n",
    "   echo \"---------------- INDEXING BAM: $FASTQ_BASE ----------------\"\n",
    "   samtools index ${STAR_OUT}/${FASTQ_BASE}_short_introns_Aligned.sortedByCoord.out.bam\n",
    "done\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/shared_space/TA_clint/trimmed_fastqs\n"
     ]
    }
   ],
   "source": [
    "echo $TRIMMED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------- FASTQC:   B_16_S26_L001_R1 ----------------\n",
      "Started analysis of B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "Analysis complete for B_16_S26_L001_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_16_S26_L001_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_16_S26_L001_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3013575\n",
      "Too short after clip: 0\n",
      "Trimmed 289890 reads by an average of 1.24 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_16_S26_L001_R1_trim ---------\n",
      "Started analysis of B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_16_S26_L001_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_16_S26_L002_R1 ----------------\n",
      "Started analysis of B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 60% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 70% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 75% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 80% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 85% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 90% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "Analysis complete for B_16_S26_L002_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_16_S26_L002_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_16_S26_L002_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2969414\n",
      "Too short after clip: 0\n",
      "Trimmed 286922 reads by an average of 1.24 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_16_S26_L002_R1_trim ---------\n",
      "Started analysis of B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_16_S26_L002_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_16_S26_L003_R1 ----------------\n",
      "Started analysis of B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 5% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 10% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 15% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 20% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 25% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 30% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 35% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 40% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 45% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 50% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 55% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 60% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 65% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 70% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 75% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 80% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 85% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 90% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Approx 95% complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "Analysis complete for B_16_S26_L003_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_16_S26_L003_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_16_S26_L003_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3024029\n",
      "Too short after clip: 0\n",
      "Trimmed 308502 reads by an average of 1.24 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_16_S26_L003_R1_trim ---------\n",
      "Started analysis of B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 50% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_16_S26_L003_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_16_S26_L004_R1 ----------------\n",
      "Started analysis of B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 5% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 10% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 15% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 20% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 25% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 30% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 35% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 40% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 45% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 50% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 55% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 60% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 65% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 70% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 75% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 80% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 85% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 90% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Approx 95% complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "Analysis complete for B_16_S26_L004_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_16_S26_L004_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_16_S26_L004_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2990234\n",
      "Too short after clip: 0\n",
      "Trimmed 307188 reads by an average of 1.24 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_16_S26_L004_R1_trim ---------\n",
      "Started analysis of B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_16_S26_L004_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_26_S10_L001_R1 ----------------\n",
      "Started analysis of B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "Analysis complete for B_26_S10_L001_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_26_S10_L001_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_26_S10_L001_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3051495\n",
      "Too short after clip: 0\n",
      "Trimmed 292800 reads by an average of 1.23 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_26_S10_L001_R1_trim ---------\n",
      "Started analysis of B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_26_S10_L001_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_26_S10_L002_R1 ----------------\n",
      "Started analysis of B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 60% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 70% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 75% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 80% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 85% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 90% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "Analysis complete for B_26_S10_L002_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_26_S10_L002_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_26_S10_L002_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3008271\n",
      "Too short after clip: 0\n",
      "Trimmed 291960 reads by an average of 1.23 bases on quality < 20\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "---------------- FASTQC:   B_26_S10_L002_R1_trim ---------\n",
      "Started analysis of B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_26_S10_L002_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_26_S10_L003_R1 ----------------\n",
      "Started analysis of B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 5% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 10% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 15% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 20% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 25% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 30% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 35% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 40% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 45% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 50% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 55% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 60% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 65% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 70% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 75% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 80% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 85% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 90% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Approx 95% complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "Analysis complete for B_26_S10_L003_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_26_S10_L003_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_26_S10_L003_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3058171\n",
      "Too short after clip: 0\n",
      "Trimmed 313450 reads by an average of 1.23 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_26_S10_L003_R1_trim ---------\n",
      "Started analysis of B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_26_S10_L003_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_26_S10_L004_R1 ----------------\n",
      "Started analysis of B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 5% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 10% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 15% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 20% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 25% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 30% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 35% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 40% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 45% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 50% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 55% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 60% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 65% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 70% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 75% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 80% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 85% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 90% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Approx 95% complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "Analysis complete for B_26_S10_L004_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_26_S10_L004_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_26_S10_L004_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3026830\n",
      "Too short after clip: 0\n",
      "Trimmed 312232 reads by an average of 1.23 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_26_S10_L004_R1_trim ---------\n",
      "Started analysis of B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_26_S10_L004_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_27_S18_L001_R1 ----------------\n",
      "Started analysis of B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for B_27_S18_L001_R1_001.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 90% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "Analysis complete for B_27_S18_L001_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_27_S18_L001_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_27_S18_L001_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2777826\n",
      "Too short after clip: 0\n",
      "Trimmed 278301 reads by an average of 1.26 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_27_S18_L001_R1_trim ---------\n",
      "Started analysis of B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_27_S18_L001_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_27_S18_L002_R1 ----------------\n",
      "Started analysis of B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 60% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 70% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 75% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 80% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 85% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 90% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "Analysis complete for B_27_S18_L002_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_27_S18_L002_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_27_S18_L002_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2745047\n",
      "Too short after clip: 0\n",
      "Trimmed 276203 reads by an average of 1.27 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_27_S18_L002_R1_trim ---------\n",
      "Started analysis of B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_27_S18_L002_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_27_S18_L003_R1 ----------------\n",
      "Started analysis of B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 5% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 10% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 15% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 20% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 25% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 30% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 35% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 40% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 45% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 50% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 55% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 60% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 65% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 70% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 75% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 80% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 85% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 90% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Approx 95% complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "Analysis complete for B_27_S18_L003_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_27_S18_L003_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_27_S18_L003_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2787481\n",
      "Too short after clip: 0\n",
      "Trimmed 296181 reads by an average of 1.27 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_27_S18_L003_R1_trim ---------\n",
      "Started analysis of B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_27_S18_L003_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_27_S18_L004_R1 ----------------\n",
      "Started analysis of B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 5% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 10% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 15% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 20% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 25% complete for B_27_S18_L004_R1_001.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 30% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 35% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 40% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 45% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 50% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 55% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 60% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 65% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 70% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 75% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 80% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 85% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 90% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Approx 95% complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "Analysis complete for B_27_S18_L004_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_27_S18_L004_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_27_S18_L004_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2755696\n",
      "Too short after clip: 0\n",
      "Trimmed 294836 reads by an average of 1.26 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_27_S18_L004_R1_trim ---------\n",
      "Started analysis of B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_27_S18_L004_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_38_S34_L001_R1 ----------------\n",
      "Started analysis of B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "Analysis complete for B_38_S34_L001_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_38_S34_L001_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_38_S34_L001_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2900454\n",
      "Too short after clip: 0\n",
      "Trimmed 296608 reads by an average of 1.25 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_38_S34_L001_R1_trim ---------\n",
      "Started analysis of B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_38_S34_L001_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_38_S34_L002_R1 ----------------\n",
      "Started analysis of B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 60% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 70% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 75% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 80% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 85% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 90% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "Analysis complete for B_38_S34_L002_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_38_S34_L002_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_38_S34_L002_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2866538\n",
      "Too short after clip: 0\n",
      "Trimmed 294344 reads by an average of 1.25 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_38_S34_L002_R1_trim ---------\n",
      "Started analysis of B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 85% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_38_S34_L002_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_38_S34_L003_R1 ----------------\n",
      "Started analysis of B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 5% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 10% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 15% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 20% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 25% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 30% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 35% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 40% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 45% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 50% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 55% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 60% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 65% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 70% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 75% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 80% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 85% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 90% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Approx 95% complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "Analysis complete for B_38_S34_L003_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_38_S34_L003_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_38_S34_L003_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2906895\n",
      "Too short after clip: 0\n",
      "Trimmed 315594 reads by an average of 1.26 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_38_S34_L003_R1_trim ---------\n",
      "Started analysis of B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_38_S34_L003_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_38_S34_L004_R1 ----------------\n",
      "Started analysis of B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 5% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 10% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 15% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 20% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 25% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 30% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 35% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 40% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 45% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 50% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 55% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 60% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 65% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 70% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 75% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 80% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 85% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 90% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Approx 95% complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "Analysis complete for B_38_S34_L004_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_38_S34_L004_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_38_S34_L004_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2883192\n",
      "Too short after clip: 0\n",
      "Trimmed 315089 reads by an average of 1.25 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_38_S34_L004_R1_trim ---------\n",
      "Started analysis of B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_38_S34_L004_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_40_S42_L001_R1 ----------------\n",
      "Started analysis of B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "Analysis complete for B_40_S42_L001_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_40_S42_L001_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_40_S42_L001_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2795366\n",
      "Too short after clip: 1\n",
      "Trimmed 276009 reads by an average of 1.25 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_40_S42_L001_R1_trim ---------\n",
      "Started analysis of B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 30% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_40_S42_L001_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_40_S42_L002_R1 ----------------\n",
      "Started analysis of B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 60% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 70% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 75% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 80% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 85% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 90% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "Analysis complete for B_40_S42_L002_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_40_S42_L002_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_40_S42_L002_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2756197\n",
      "Too short after clip: 0\n",
      "Trimmed 273052 reads by an average of 1.25 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_40_S42_L002_R1_trim ---------\n",
      "Started analysis of B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_40_S42_L002_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_40_S42_L003_R1 ----------------\n",
      "Started analysis of B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 5% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 10% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 15% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 20% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 25% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 30% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 35% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 40% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 45% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 50% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 55% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 60% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 65% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 70% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 75% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 80% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 85% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 90% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Approx 95% complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "Analysis complete for B_40_S42_L003_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_40_S42_L003_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_40_S42_L003_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2798996\n",
      "Too short after clip: 0\n",
      "Trimmed 292453 reads by an average of 1.26 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_40_S42_L003_R1_trim ---------\n",
      "Started analysis of B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_40_S42_L003_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_40_S42_L004_R1 ----------------\n",
      "Started analysis of B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 5% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 10% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 15% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 20% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 25% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 30% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 35% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 40% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 45% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 50% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 55% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 60% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 65% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 70% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 75% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 80% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 85% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 90% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Approx 95% complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "Analysis complete for B_40_S42_L004_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_40_S42_L004_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_40_S42_L004_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_40_S42_L004_R1_001.trim.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 2761227\n",
      "Too short after clip: 0\n",
      "Trimmed 291394 reads by an average of 1.25 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_40_S42_L004_R1_trim ---------\n",
      "Started analysis of B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_40_S42_L004_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_4_S2_L001_R1 ----------------\n",
      "Started analysis of B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "Analysis complete for B_4_S2_L001_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_4_S2_L001_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_4_S2_L001_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3096747\n",
      "Too short after clip: 0\n",
      "Trimmed 304915 reads by an average of 1.24 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_4_S2_L001_R1_trim ---------\n",
      "Started analysis of B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_4_S2_L001_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_4_S2_L002_R1 ----------------\n",
      "Started analysis of B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 60% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 70% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 75% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 80% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 85% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 90% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "Analysis complete for B_4_S2_L002_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_4_S2_L002_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_4_S2_L002_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3041813\n",
      "Too short after clip: 0\n",
      "Trimmed 301017 reads by an average of 1.24 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_4_S2_L002_R1_trim ---------\n",
      "Started analysis of B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_4_S2_L002_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_4_S2_L003_R1 ----------------\n",
      "Started analysis of B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 5% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 10% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 15% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 20% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 25% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 30% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 35% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 40% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 45% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 50% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 55% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 60% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 65% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 70% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 75% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 80% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 85% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Approx 90% complete for B_4_S2_L003_R1_001.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 95% complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "Analysis complete for B_4_S2_L003_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_4_S2_L003_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_4_S2_L003_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3112341\n",
      "Too short after clip: 0\n",
      "Trimmed 327032 reads by an average of 1.25 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_4_S2_L003_R1_trim ---------\n",
      "Started analysis of B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_4_S2_L003_R1_001.trim.fastq.gz\n",
      "\n",
      "---------------- FASTQC:   B_4_S2_L004_R1 ----------------\n",
      "Started analysis of B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 5% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 10% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 15% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 20% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 25% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 30% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 35% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 40% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 45% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 50% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 55% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 60% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 65% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 70% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 75% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 80% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 85% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 90% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Approx 95% complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "Analysis complete for B_4_S2_L004_R1_001.fastq.gz\n",
      "\n",
      "---------------- TRIMMING: B_4_S2_L004_R1 ----------------\n",
      "Command Line: /shared_space/TA_clint/myinfo/neb_e7600_adapters.fasta /data/hts2018_course/Granek_5003_180726A5/B_4_S2_L004_R1_001.fastq.gz -q 20 -x 0.5 -o /shared_space/TA_clint/trimmed_fastqs/B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Files: 1\n",
      "Total reads: 3057930\n",
      "Too short after clip: 0\n",
      "Trimmed 323252 reads by an average of 1.24 bases on quality < 20\n",
      "\n",
      "---------------- FASTQC:   B_4_S2_L004_R1_trim ---------\n",
      "Started analysis of B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 5% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 10% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 15% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 20% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 25% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 30% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 35% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 40% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 45% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 50% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 55% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 60% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 65% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 70% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 75% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 80% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 85% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 90% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Approx 95% complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "Analysis complete for B_4_S2_L004_R1_001.trim.fastq.gz\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for FASTQ in $(ls $RAW_FASTQS | grep ^B); do\n",
    "    FASTQ_BASE=$(basename ${FASTQ} '_001.fastq.gz')\n",
    "    \n",
    "    ##################################################\n",
    "    echo \"---------------- FASTQC:   ${FASTQ_BASE} ----------------\"\n",
    "    fastqc --extract ${RAW_FASTQS}/${FASTQ_BASE}_001.fastq.gz -o $QC\n",
    "    echo \"\"\n",
    "    \n",
    "    echo \"---------------- TRIMMING: ${FASTQ_BASE} ----------------\"\n",
    "    fastq-mcf \\\n",
    "        $ADAPTERS \\\n",
    "        ${RAW_FASTQS}/${FASTQ_BASE}_001.fastq.gz \\\n",
    "        -q 20 -x 0.5 \\\n",
    "        -o ${TRIMMED}/${FASTQ_BASE}_001.trim.fastq.gz\n",
    "    echo \"\"\n",
    "    \n",
    "    echo \"---------------- FASTQC:   ${FASTQ_BASE}_trim ---------\"\n",
    "    fastqc --extract ${TRIMMED}/${FASTQ_BASE}_001.trim.fastq.gz -o $QC\n",
    "    echo \"\"  \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;31mB_16_S26_L001_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_38_S34_L001_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_16_S26_L002_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_38_S34_L002_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_16_S26_L003_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_38_S34_L003_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_16_S26_L004_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_38_S34_L004_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_26_S10_L001_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_40_S42_L001_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_26_S10_L002_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_40_S42_L002_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_26_S10_L003_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_40_S42_L003_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_26_S10_L004_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_40_S42_L004_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_27_S18_L001_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_4_S2_L001_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_27_S18_L002_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_4_S2_L002_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_27_S18_L003_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_4_S2_L003_R1_001.trim.fastq.gz\u001b[0m\n",
      "\u001b[01;31mB_27_S18_L004_R1_001.trim.fastq.gz\u001b[0m  \u001b[01;31mB_4_S2_L004_R1_001.trim.fastq.gz\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls $TRIMMED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------- STAR:     B_16_S26_L001_R1 ---------\n",
      "Aug 05 03:18:00 ..... started STAR run\n",
      "Aug 05 03:18:00 ..... loading genome\n",
      "Aug 05 03:18:05 ..... started mapping\n",
      "Aug 05 03:19:04 ..... started sorting BAM\n",
      "Aug 05 03:19:15 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_16_S26_L002_R1 ---------\n",
      "Aug 05 03:19:15 ..... started STAR run\n",
      "Aug 05 03:19:15 ..... loading genome\n",
      "Aug 05 03:19:21 ..... started mapping\n",
      "Aug 05 03:20:16 ..... started sorting BAM\n",
      "Aug 05 03:20:26 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_16_S26_L003_R1 ---------\n",
      "Aug 05 03:20:26 ..... started STAR run\n",
      "Aug 05 03:20:26 ..... loading genome\n",
      "Aug 05 03:20:27 ..... started mapping\n",
      "Aug 05 03:21:22 ..... started sorting BAM\n",
      "Aug 05 03:21:33 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_16_S26_L004_R1 ---------\n",
      "Aug 05 03:21:33 ..... started STAR run\n",
      "Aug 05 03:21:33 ..... loading genome\n",
      "Aug 05 03:21:34 ..... started mapping\n",
      "Aug 05 03:22:31 ..... started sorting BAM\n",
      "Aug 05 03:22:42 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_26_S10_L001_R1 ---------\n",
      "Aug 05 03:22:42 ..... started STAR run\n",
      "Aug 05 03:22:42 ..... loading genome\n",
      "Aug 05 03:22:43 ..... started mapping\n",
      "Aug 05 03:23:38 ..... started sorting BAM\n",
      "Aug 05 03:23:50 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_26_S10_L002_R1 ---------\n",
      "Aug 05 03:23:50 ..... started STAR run\n",
      "Aug 05 03:23:50 ..... loading genome\n",
      "Aug 05 03:23:51 ..... started mapping\n",
      "Aug 05 03:24:44 ..... started sorting BAM\n",
      "Aug 05 03:24:56 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_26_S10_L003_R1 ---------\n",
      "Aug 05 03:24:56 ..... started STAR run\n",
      "Aug 05 03:24:56 ..... loading genome\n",
      "Aug 05 03:24:56 ..... started mapping\n",
      "Aug 05 03:25:52 ..... started sorting BAM\n",
      "Aug 05 03:26:04 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_26_S10_L004_R1 ---------\n",
      "Aug 05 03:26:04 ..... started STAR run\n",
      "Aug 05 03:26:04 ..... loading genome\n",
      "Aug 05 03:26:04 ..... started mapping\n",
      "Aug 05 03:26:59 ..... started sorting BAM\n",
      "Aug 05 03:27:11 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_27_S18_L001_R1 ---------\n",
      "Aug 05 03:27:11 ..... started STAR run\n",
      "Aug 05 03:27:11 ..... loading genome\n",
      "Aug 05 03:27:11 ..... started mapping\n",
      "Aug 05 03:28:22 ..... started sorting BAM\n",
      "Aug 05 03:28:32 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_27_S18_L002_R1 ---------\n",
      "Aug 05 03:28:32 ..... started STAR run\n",
      "Aug 05 03:28:32 ..... loading genome\n",
      "Aug 05 03:28:32 ..... started mapping\n",
      "Aug 05 03:29:40 ..... started sorting BAM\n",
      "Aug 05 03:29:49 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_27_S18_L003_R1 ---------\n",
      "Aug 05 03:29:50 ..... started STAR run\n",
      "Aug 05 03:29:50 ..... loading genome\n",
      "Aug 05 03:29:50 ..... started mapping\n",
      "Aug 05 03:30:59 ..... started sorting BAM\n",
      "Aug 05 03:31:09 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_27_S18_L004_R1 ---------\n",
      "Aug 05 03:31:09 ..... started STAR run\n",
      "Aug 05 03:31:09 ..... loading genome\n",
      "Aug 05 03:31:09 ..... started mapping\n",
      "Aug 05 03:32:20 ..... started sorting BAM\n",
      "Aug 05 03:32:30 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_38_S34_L001_R1 ---------\n",
      "Aug 05 03:32:30 ..... started STAR run\n",
      "Aug 05 03:32:30 ..... loading genome\n",
      "Aug 05 03:32:31 ..... started mapping\n",
      "Aug 05 03:33:29 ..... started sorting BAM\n",
      "Aug 05 03:33:38 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_38_S34_L002_R1 ---------\n",
      "Aug 05 03:33:39 ..... started STAR run\n",
      "Aug 05 03:33:39 ..... loading genome\n",
      "Aug 05 03:33:39 ..... started mapping\n",
      "Aug 05 03:34:36 ..... started sorting BAM\n",
      "Aug 05 03:34:45 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_38_S34_L003_R1 ---------\n",
      "Aug 05 03:34:46 ..... started STAR run\n",
      "Aug 05 03:34:46 ..... loading genome\n",
      "Aug 05 03:34:46 ..... started mapping\n",
      "Aug 05 03:35:44 ..... started sorting BAM\n",
      "Aug 05 03:35:54 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_38_S34_L004_R1 ---------\n",
      "Aug 05 03:35:54 ..... started STAR run\n",
      "Aug 05 03:35:54 ..... loading genome\n",
      "Aug 05 03:35:54 ..... started mapping\n",
      "Aug 05 03:36:52 ..... started sorting BAM\n",
      "Aug 05 03:37:02 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_40_S42_L001_R1 ---------\n",
      "Aug 05 03:37:03 ..... started STAR run\n",
      "Aug 05 03:37:03 ..... loading genome\n",
      "Aug 05 03:37:03 ..... started mapping\n",
      "Aug 05 03:38:01 ..... started sorting BAM\n",
      "Aug 05 03:38:10 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_40_S42_L002_R1 ---------\n",
      "Aug 05 03:38:11 ..... started STAR run\n",
      "Aug 05 03:38:11 ..... loading genome\n",
      "Aug 05 03:38:11 ..... started mapping\n",
      "Aug 05 03:39:09 ..... started sorting BAM\n",
      "Aug 05 03:39:19 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_40_S42_L003_R1 ---------\n",
      "Aug 05 03:39:19 ..... started STAR run\n",
      "Aug 05 03:39:19 ..... loading genome\n",
      "Aug 05 03:39:19 ..... started mapping\n",
      "Aug 05 03:40:19 ..... started sorting BAM\n",
      "Aug 05 03:40:29 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_40_S42_L004_R1 ---------\n",
      "Aug 05 03:40:29 ..... started STAR run\n",
      "Aug 05 03:40:29 ..... loading genome\n",
      "Aug 05 03:40:29 ..... started mapping\n",
      "Aug 05 03:41:27 ..... started sorting BAM\n",
      "Aug 05 03:41:36 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_4_S2_L001_R1 ---------\n",
      "Aug 05 03:41:37 ..... started STAR run\n",
      "Aug 05 03:41:37 ..... loading genome\n",
      "Aug 05 03:41:37 ..... started mapping\n",
      "Aug 05 03:44:36 ..... started sorting BAM\n",
      "Aug 05 03:44:44 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_4_S2_L002_R1 ---------\n",
      "Aug 05 03:44:44 ..... started STAR run\n",
      "Aug 05 03:44:44 ..... loading genome\n",
      "Aug 05 03:44:45 ..... started mapping\n",
      "Aug 05 03:47:41 ..... started sorting BAM\n",
      "Aug 05 03:47:49 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_4_S2_L003_R1 ---------\n",
      "Aug 05 03:47:49 ..... started STAR run\n",
      "Aug 05 03:47:49 ..... loading genome\n",
      "Aug 05 03:47:49 ..... started mapping\n",
      "Aug 05 03:50:47 ..... started sorting BAM\n",
      "Aug 05 03:50:56 ..... finished successfully\n",
      "\n",
      "---------------- STAR:     B_4_S2_L004_R1 ---------\n",
      "Aug 05 03:50:56 ..... started STAR run\n",
      "Aug 05 03:50:56 ..... loading genome\n",
      "Aug 05 03:50:56 ..... started mapping\n",
      "Aug 05 03:53:53 ..... started sorting BAM\n",
      "Aug 05 03:54:01 ..... finished successfully\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for FASTQ in $(ls $RAW_FASTQS | grep ^B); do\n",
    "    FASTQ_BASE=$(basename ${FASTQ} '_001.fastq.gz')\n",
    "    echo \"---------------- STAR:     ${FASTQ_BASE} ---------\"\n",
    "    STAR \\\n",
    "        --runMode           alignReads    \\\n",
    "        --twopassMode       None          \\\n",
    "        --genomeDir         ${GENOME_DIR} \\\n",
    "        --readFilesIn       ${TRIMMED}/${FASTQ_BASE}_001.trim.fastq.gz   \\\n",
    "        --readFilesCommand  gunzip -c                                \\\n",
    "        --outFileNamePrefix ${STAR_OUT}/${FASTQ_BASE}_short_introns_ \\\n",
    "        --quantMode         GeneCounts              \\\n",
    "        --outSAMtype        BAM SortedByCoordinate  \\\n",
    "        --alignIntronMax    5000                    \\\n",
    "        --outSJfilterIntronMaxVsReadN 500 1000 2000\n",
    "    echo \"\"\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/shared_space/TA_clint/star_out/B_16_S26_L001_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_16_S26_L002_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_16_S26_L003_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_16_S26_L004_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_26_S10_L001_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_26_S10_L002_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_26_S10_L003_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_26_S10_L004_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_27_S18_L001_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_27_S18_L002_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_27_S18_L003_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_27_S18_L004_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_38_S34_L001_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_38_S34_L002_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_38_S34_L003_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_38_S34_L004_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_40_S42_L001_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_40_S42_L002_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_40_S42_L003_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_40_S42_L004_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_4_S2_L001_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_4_S2_L002_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_4_S2_L003_R1_short_introns_Aligned.sortedByCoord.out.bam\n",
      "/shared_space/TA_clint/star_out/B_4_S2_L004_R1_short_introns_Aligned.sortedByCoord.out.bam\n"
     ]
    }
   ],
   "source": [
    "for BAM in ${STAR_OUT}/*_Aligned.sortedByCoord.out.bam; do\n",
    "    echo $BAM\n",
    "    samtools index $BAM\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check the output"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`awk '{$1=$1;print}'`: [trim the leading whitespaces](https://unix.stackexchange.com/questions/102008/how-do-i-trim-leading-and-trailing-whitespace-from-each-line-of-some-output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "B_16_S26_L001_R1\n",
      "Number of input reads | 3013575\n",
      "\n",
      "B_16_S26_L002_R1\n",
      "Number of input reads | 2969414\n",
      "\n",
      "B_16_S26_L003_R1\n",
      "Number of input reads | 3024029\n",
      "\n",
      "B_16_S26_L004_R1\n",
      "Number of input reads | 2990234\n",
      "\n",
      "B_26_S10_L001_R1\n",
      "Number of input reads | 3051495\n",
      "\n",
      "B_26_S10_L002_R1\n",
      "Number of input reads | 3008271\n",
      "\n",
      "B_26_S10_L003_R1\n",
      "Number of input reads | 3058171\n",
      "\n",
      "B_26_S10_L004_R1\n",
      "Number of input reads | 3026830\n",
      "\n",
      "B_27_S18_L001_R1\n",
      "Number of input reads | 2777826\n",
      "\n",
      "B_27_S18_L002_R1\n",
      "Number of input reads | 2745047\n",
      "\n",
      "B_27_S18_L003_R1\n",
      "Number of input reads | 2787481\n",
      "\n",
      "B_27_S18_L004_R1\n",
      "Number of input reads | 2755696\n",
      "\n",
      "B_38_S34_L001_R1\n",
      "Number of input reads | 2900454\n",
      "\n",
      "B_38_S34_L002_R1\n",
      "Number of input reads | 2866538\n",
      "\n",
      "B_38_S34_L003_R1\n",
      "Number of input reads | 2906895\n",
      "\n",
      "B_38_S34_L004_R1\n",
      "Number of input reads | 2883192\n",
      "\n",
      "B_40_S42_L001_R1\n",
      "Number of input reads | 2795365\n",
      "\n",
      "B_40_S42_L002_R1\n",
      "Number of input reads | 2756197\n",
      "\n",
      "B_40_S42_L003_R1\n",
      "Number of input reads | 2798996\n",
      "\n",
      "B_40_S42_L004_R1\n",
      "Number of input reads | 2761227\n",
      "\n",
      "B_4_S2_L001_R1\n",
      "Number of input reads | 3096747\n",
      "\n",
      "B_4_S2_L002_R1\n",
      "Number of input reads | 3041813\n",
      "\n",
      "B_4_S2_L003_R1\n",
      "Number of input reads | 3112341\n",
      "\n",
      "B_4_S2_L004_R1\n",
      "Number of input reads | 3057930\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for file in ${STAR_OUT}/*_short_introns_Log.final.out; do\n",
    "    echo $(basename ${file} \"_short_introns_Log.final.out\") \n",
    "    cat ${file} | grep \"Number of input reads\" | awk '{$1=$1;print}'\n",
    "    echo \"\"\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "B_16_S26_L001_R1_short_introns\n",
      "N_unmapped\t9159\t9159\t9159\n",
      "N_multimapping\t470249\t470249\t470249\n",
      "N_noFeature\t469318\t2453101\t476341\n",
      "\n",
      "B_16_S26_L002_R1_short_introns\n",
      "N_unmapped\t8626\t8626\t8626\n",
      "N_multimapping\t463192\t463192\t463192\n",
      "N_noFeature\t462845\t2417444\t469769\n",
      "\n",
      "B_16_S26_L003_R1_short_introns\n",
      "N_unmapped\t9416\t9416\t9416\n",
      "N_multimapping\t469193\t469193\t469193\n",
      "N_noFeature\t472165\t2463810\t479098\n",
      "\n",
      "B_16_S26_L004_R1_short_introns\n",
      "N_unmapped\t9032\t9032\t9032\n",
      "N_multimapping\t463449\t463449\t463449\n",
      "N_noFeature\t468979\t2437000\t475947\n",
      "\n",
      "B_26_S10_L001_R1_short_introns\n",
      "N_unmapped\t12335\t12335\t12335\n",
      "N_multimapping\t988897\t988897\t988897\n",
      "N_noFeature\t199991\t2023246\t205465\n",
      "\n",
      "B_26_S10_L002_R1_short_introns\n",
      "N_unmapped\t11855\t11855\t11855\n",
      "N_multimapping\t975103\t975103\t975103\n",
      "N_noFeature\t197722\t1994270\t203273\n",
      "\n",
      "B_26_S10_L003_R1_short_introns\n",
      "N_unmapped\t12603\t12603\t12603\n",
      "N_multimapping\t986675\t986675\t986675\n",
      "N_noFeature\t200730\t2031542\t206116\n",
      "\n",
      "B_26_S10_L004_R1_short_introns\n",
      "N_unmapped\t12128\t12128\t12128\n",
      "N_multimapping\t975201\t975201\t975201\n",
      "N_noFeature\t200920\t2012640\t206384\n",
      "\n",
      "B_27_S18_L001_R1_short_introns\n",
      "N_unmapped\t135943\t135943\t135943\n",
      "N_multimapping\t235122\t235122\t235122\n",
      "N_noFeature\t764659\t2198033\t858777\n",
      "\n",
      "B_27_S18_L002_R1_short_introns\n",
      "N_unmapped\t134465\t134465\t134465\n",
      "N_multimapping\t232656\t232656\t232656\n",
      "N_noFeature\t758510\t2171614\t851671\n",
      "\n",
      "B_27_S18_L003_R1_short_introns\n",
      "N_unmapped\t136793\t136793\t136793\n",
      "N_multimapping\t235656\t235656\t235656\n",
      "N_noFeature\t768195\t2205702\t861996\n",
      "\n",
      "B_27_S18_L004_R1_short_introns\n",
      "N_unmapped\t135500\t135500\t135500\n",
      "N_multimapping\t232671\t232671\t232671\n",
      "N_noFeature\t763643\t2181252\t856313\n",
      "\n",
      "B_38_S34_L001_R1_short_introns\n",
      "N_unmapped\t20735\t20735\t20735\n",
      "N_multimapping\t160735\t160735\t160735\n",
      "N_noFeature\t1025939\t2581153\t1040336\n",
      "\n",
      "B_38_S34_L002_R1_short_introns\n",
      "N_unmapped\t20100\t20100\t20100\n",
      "N_multimapping\t159079\t159079\t159079\n",
      "N_noFeature\t1017347\t2551131\t1031678\n",
      "\n",
      "B_38_S34_L003_R1_short_introns\n",
      "N_unmapped\t21011\t21011\t21011\n",
      "N_multimapping\t160959\t160959\t160959\n",
      "N_noFeature\t1029968\t2587188\t1044240\n",
      "\n",
      "B_38_S34_L004_R1_short_introns\n",
      "N_unmapped\t20597\t20597\t20597\n",
      "N_multimapping\t159614\t159614\t159614\n",
      "N_noFeature\t1025716\t2565982\t1040103\n",
      "\n",
      "B_40_S42_L001_R1_short_introns\n",
      "N_unmapped\t36532\t36532\t36532\n",
      "N_multimapping\t130184\t130184\t130184\n",
      "N_noFeature\t777671\t2458488\t802513\n",
      "\n",
      "B_40_S42_L002_R1_short_introns\n",
      "N_unmapped\t35255\t35255\t35255\n",
      "N_multimapping\t128521\t128521\t128521\n",
      "N_noFeature\t771045\t2424776\t795515\n",
      "\n",
      "B_40_S42_L003_R1_short_introns\n",
      "N_unmapped\t36591\t36591\t36591\n",
      "N_multimapping\t130261\t130261\t130261\n",
      "N_noFeature\t778838\t2461864\t803866\n",
      "\n",
      "B_40_S42_L004_R1_short_introns\n",
      "N_unmapped\t36122\t36122\t36122\n",
      "N_multimapping\t128961\t128961\t128961\n",
      "N_noFeature\t773000\t2428015\t797623\n",
      "\n",
      "B_4_S2_L001_R1_short_introns\n",
      "N_unmapped\t1017930\t1017930\t1017930\n",
      "N_multimapping\t610733\t610733\t610733\n",
      "N_noFeature\t114401\t588455\t962478\n",
      "\n",
      "B_4_S2_L002_R1_short_introns\n",
      "N_unmapped\t998404\t998404\t998404\n",
      "N_multimapping\t601446\t601446\t601446\n",
      "N_noFeature\t112318\t576305\t947034\n",
      "\n",
      "B_4_S2_L003_R1_short_introns\n",
      "N_unmapped\t1023777\t1023777\t1023777\n",
      "N_multimapping\t612603\t612603\t612603\n",
      "N_noFeature\t114804\t591776\t967261\n",
      "\n",
      "B_4_S2_L004_R1_short_introns\n",
      "N_unmapped\t1006517\t1006517\t1006517\n",
      "N_multimapping\t599896\t599896\t599896\n",
      "N_noFeature\t113419\t581450\t952797\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for file in ${STAR_OUT}/*ReadsPerGene.out.tab; do\n",
    "    echo $(basename ${file} \"_ReadsPerGene.out.tab\") \n",
    "    head -3 ${file}\n",
    "    echo \"\"\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "for FASTQ in $RAW_FASTQS/27_MA_P_S38_L00[1]_R1_001.fastq.gz\n",
    "    do\n",
    "        FASTQ_BASE=\"$(basename ${FASTQ} '_001.fastq.gz')\"\n",
    "        echo \"---------------- TRIMMING: $FASTQ_BASE ----------------\"\n",
    "        fastq-mcf \\\n",
    "            $MYINFO/neb_e7600_adapters.fasta \\\n",
    "            $RAW_FASTQS/${FASTQ_BASE}_001.fastq.gz \\\n",
    "            -q 20 -x 0.5 \\\n",
    "            -o $TRIMMED/${FASTQ_BASE}_001.trim.fastq.gz\n",
    "        \n",
    "        echo \"---------------- MAPPING: $FASTQ_BASE ----------------\"\n",
    "        STAR \\\n",
    "            --runMode alignReads \\\n",
    "            --twopassMode None \\\n",
    "            --genomeDir $GENOME_DIR \\\n",
    "            --readFilesIn $TRIMMED/${FASTQ_BASE}_001.trim.fastq.gz \\\n",
    "            --readFilesCommand gunzip -c \\\n",
    "            --outFileNamePrefix ${STAR_OUT}/${FASTQ_BASE}_short_introns_ \\\n",
    "            --quantMode GeneCounts \\\n",
    "            --outSAMtype BAM SortedByCoordinate \\\n",
    "            --alignIntronMax 5000 \\\n",
    "            --outSJfilterIntronMaxVsReadN 500 1000 2000 \\\n",
    "            \n",
    "        echo \"---------------- INDEXING BAM: $FASTQ_BASE ----------------\"\n",
    "        samtools index ${STAR_OUT}/${FASTQ_BASE}_short_introns_Aligned.sortedByCoord.out.bam\n",
    "    done \n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
